Sources/FluentKit/Query/Builder/QueryBuilder.swift

```diff
@@ -229,12 +234,29 @@ public final class QueryBuilder<Model>
                 }
             }
         }
-        
-        // If deleted models aren't included, add filters
-        // to exclude them for each model being queried.
-        if !self.includeDeleted {
-            for model in self.models {
-                model.excludeDeleted(from: &query)
+
+        for model in self.models {
+            let initializedModel = model.init()
+            
+            // If deleted models aren't included, add filters
+            // to exclude them for each model being queried.
+            if !self.includeDeleted {
+                model.excludeDeleted(initialized: initializedModel, from: &query)
+            }
+            
+            // If we don't want to force delete, just touch
+            // the timestamp model. We also change it to an
+            // update query. If it is a force delete, still
+            // touch the timestamps
+            if !self.forceDelete, let _ = initializedModel.deletedTimestamp {
+                initializedModel.touchTimestamps(.update, .delete)
+                query.action = .update
+                query.input = [.dictionary(initializedModel.input.values)]
+            } else {
+                switch query.action {
+                    case .create: initializedModel.touchTimestamps(.create, .update)
```

Comment:
I don't think this will actually do anything. We should have some tests for it to be sure. You'll need to add the timestamp fields the existing values in `query.input`. This needs to work for both normal and batch saves. I'm also realizing now that once this is added here, we should be able to remove timestamp logic from the `Model+CRUD` file since it will be handled entirely here. That's a lot better 👍